src/pspm_downsample.m

```diff
@@ -23,10 +24,42 @@
 %% 2 Check input arguments
 if nargin < 2
   warning('Not enough input arguments.'); return
-elseif floor(freqratio) ~= freqratio
-  warning('Frequency ratio must be integer.'); return
+% elseif floor(freqratio) ~= freqratio
+%   warning('Frequency ratio must be integer.'); return
 end
 %% 3 Performing downsampling
-data = data(freqratio:freqratio:end);
-sts = 1;
+
+freqratio = sr/sr_down;
+
+
+if freqratio == ceil(freqratio) % NB isinteger might not work for some values
```

Comment:
I'd suggest exchanging the first and second case, i.e. start with `if settings.signal`. This way, we can be sure that an in-built function handles most of the cases, even when frequency ratio is integer.

Reply:
But will the resampled data not be a bit off (in the case of integer ratios)?